src/query/ts/m3db/convert.go

```diff
@@ -54,31 +57,56 @@ func (b blockReplicas) Less(i, j int) bool {
 	return b[i].start.Before(b[j].start)
 }
 
-// ConvertM3DBSeriesIterators converts m3db SeriesIterators to SeriesBlocks
-// which are used to construct Blocks for query processing.
-func ConvertM3DBSeriesIterators(iterators encoding.SeriesIterators, iterAlloc encoding.ReaderIteratorAllocate) ([]SeriesBlocks, error) {
-	defer iterators.Close()
-	multiSeriesBlocks := make([]SeriesBlocks, iterators.Len())
+// ConvertM3DBSeriesIterators converts series iterators to iterator blocks
+func ConvertM3DBSeriesIterators(
```

Comment:
when would you have multiple blocks from series iterators ?

Reply:
When I enable splitting series iterators into block iterators, this will return multiple blocks.

Reply:
maybe add that TODO?

Reply:
It's the next part of this, doesn't really need a todo since it should be fixed soon